src/replication/model/registry.go

```diff
@@ -124,3 +176,32 @@ type RegistryInfo struct {
 	SupportedResourceFilters []*FilterStyle `json:"supported_resource_filters"`
 	SupportedTriggers        []TriggerType  `json:"supported_triggers"`
 }
+
+// AdapterPattern provides base info and capability declarations of the registry
+type AdapterPattern struct {
+	EndpointPattern   *EndpointPattern   `json:"endpoint_pattern"`
```

Comment:
I think we can use a list here for endpoints, for docker hub it contains only one item, and for ali it contains multiple items. It's easy for frontend as they use the same logic to render the UI